### PR TITLE
Make agents able to connect to an existing DCA

### PR DESCRIPTION
**Pull request by hkaj**
#### What this PR does / why we need it:

Make agents able to connect to an existing DCA

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

the following `values.yaml` file can be use to instantiate `datadog` chart on windows nodes.

```yaml
targetSystem: windows

existingClusterAgent:
  join: true
  serviceName: dd-datadog-cluster-agent
  tokenSecretName: dd-datadog-cluster-agent

datadog-crds:
  crds:
    # datadog-crds.crds.datadogMetrics -- Set to true to deploy the DatadogMetrics CRD
    datadogMetrics: false

# Disable kube-state-metrics deployement
datadog:
  kubeStateMetricsEnabled: false
```

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has beed updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/values.yaml [hunk 1]
```diff
@@ -396,8 +396,11 @@ clusterAgent:
   # clusterAgent.token -- Cluster Agent token is a preshared key between node agents and cluster agent (autogenerated if empty, needs to be at least 32 characters a-zA-z)
   token: ""
 
-  # clusterAgent.tokenExistingSecret -- Existing secret name to use for Cluster Agent token
-  tokenExistingSecret: ""
+  # clusterAgent.tokenExistingSecret -- Existing secret name to use for Cluster Agent token, useful if you have a Cluster Agent deployed outside of this chart release
+  tokenExistingSecret:  # <EXISTING_DCA_SECRET_NAME>
+
+  # clusterAgent.existingService -- Existing service name to use for reaching the Cluster Agent, useful if you have a Cluster Agent deployed outside of this chart release
+  existingService:  # <EXISTING_DCA_SERVICE_NAME>
```
Comment (clamoriniere) [inline charts/datadog/values.yaml:403]: maybe we can create a specific section:
```yaml
config:
  existingClusterAgent:
     join: false
     serviceName: "cluster-agent"
     tokenSecretName: ""
```

wdyt?

Comment (hkaj) [reply charts/datadog/values.yaml:403]: yeah for sure, but be aware that i haven't tested it for every single feature

File: charts/datadog/templates/_helpers.tpl [hunk 2]
```diff
@@ -280,11 +280,46 @@ false
...
+{{- define "existingClusterAgent-configured" -}}
+{{- if and .Values.existingClusterAgent.join .Values.existingClusterAgent.serviceName .Values.existingClusterAgent.tokenSecretName -}}
+true
+{{- else -}}
+false
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if the ClusterAgent needs to be deployed
```
Comment (hkaj) [inline charts/datadog/templates/_helpers.tpl:295]: if `existingClusterAgent-configured` the cluster agent doesn't need to be deployed. This flag is more about configuring connection to the cluster agent no?

Comment (hkaj) [reply charts/datadog/templates/_helpers.tpl:295]: cc @clamoriniere

Comment (clamoriniere) [reply charts/datadog/templates/_helpers.tpl:295]: it is to know if the agent needs to contact the cluster-agent

Comment (clamoriniere) [reply charts/datadog/templates/_helpers.tpl]: cc: @hkaj done

File: charts/datadog/templates/_containers-common-env.yaml [hunk 3]
```diff
@@ -100,14 +100,22 @@ Return a list of env-vars if the cluster-agent is enabled
...
+{{- else if eq (include "existingClusterAgent-configured" .) "true" }}
+- name: DD_CLUSTER_AGENT_KUBERNETES_SERVICE_NAME
+  value: {{ .Values.existingClusterAgent.serviceName | quote }}
+- name: DD_CLUSTER_AGENT_AUTH_TOKEN
+  valueFrom:
+    secretKeyRef:
+        name: {{ .Values.existingClusterAgent.tokenSecretName | quote }}
+        key: token
 {{- end }}
 {{- end -}}
```
Comment (xornivore) [inline charts/datadog/templates/_containers-common-env.yaml:121]: Nit.

**Review by xornivore — APPROVED** [review]

**Pull request merged**